test/FunctionalTests/Microsoft.OData.Core.Tests/ODataMessageWriterTests.cs

```diff
@@ -942,7 +959,19 @@ public async Task WriteMetadataDocumentAsync_WorksForXmlCsdl()
             // Act - XML
             string payload = await this.WriteAndGetPayloadAsync(edmModel, "application/xml", async omWriter =>
             {
-                await omWriter.WriteMetadataDocumentAsync();
+                try
+                {
+                    await omWriter.WriteMetadataDocumentAsync();
+                }
+                catch (Exception e)

```

Comment:
We should catch _specific_ exception here if possible

Reply:
See: https://github.com/OData/odata.net/pull/2683/files#r1228980286